DOC: Add constructor parameter documentation to FFMpegFileWriter

nrnavaneet opened the pull request:
## Summary

Closes #22831

This PR improves the documentation for `matplotlib.animation.FFMpegFileWriter` by explicitly listing the constructor parameters (`fps`, `codec`, `bitrate`, `extra_args`, and `metadata`) in the class docstring.

## Why this matters

The class currently inherits these parameters via `*args` and `**kwargs`, making them invisible in the Sphinx-generated documentation. By adding the parameters directly to the class docstring following the NumPy docstring standard, users will now be able to see and understand how to use the writer more effectively.

## What was done

- Added constructor parameter documentation to the `FFMpegFileWriter` class docstring in `animation.py`
- Maintained existing explanatory content and references in the docstring
- Rebuilt the documentation locally to verify changes render correctly

## Additional Notes

If accepted, the same approach could be applied to similar classes like `ImageMagickFileWriter` that suffer from the same inheritance visibility issue.

QuLogic commented on lib/matplotlib/animation.py:
```diff
@@ -636,6 +638,9 @@ def _args(self):
         return [self.bin_path(), *args, *self.output_args]
 
+# Workaround: Set class docstring from __init__ for Sphinx visibility
+FFMpegFileWriter.__doc__ = inspect.getdoc(FFMpegFileWriter.__init__)
```
This doesn't look right; it's replacing the docstring that's already there.

timhoffm replied:
This actually works, because `inspect.getdoc` can resolve this, only Sphinx cannot. See https://github.com/matplotlib/matplotlib/issues/22831#issuecomment-3019698791. The proper fix would be to report this upstream to Sphinx, but I don’t have the time right now.

QuLogic replied:
To be more specific, @timhoffm suggested setting `FFMpegFileWriter.__init__.__doc__`, but you are setting `FFMpegFileWriter.__doc__`.

QuLogic commented on lib/matplotlib/animation.py:
```diff
@@ -615,6 +616,7 @@ class FFMpegFileWriter(FFMpegBase, FileMovieWriter):
     """
     supported_formats = ['png', 'jpeg', 'tiff', 'raw', 'rgba']
 
+
```
Extra change.

nrnavaneet replied:
Thank you for the change.

nrnavaneet commented:
Hi @timhoffm @QuLogic — I've removed the dynamic `__doc__` assignment and added a full NumPy-style docstring directly to the `__init__` method. This keeps the class docstring intact while exposing constructor parameters clearly to Sphinx.

Let me know if you'd like any further adjustments!

timhoffm commented:
As said [above](https://github.com/matplotlib/matplotlib/pull/30222#issuecomment-3019669406), I'm not in favor of copy-pasting the docstring. If the programmatic copy workaround does not work (I've not fully followed @QuLogic's argument), then let's rather see whether we can fix the root cause and get sphinx to correctly deduce the init docstring. I've opened an issue to sphinx: https://github.com/sphinx-doc/sphinx/issues/13721
